stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -255,6 +255,12 @@ adjoint(B::Bidiagonal) = Adjoint(B)
 transpose(B::Bidiagonal) = Transpose(B)
 adjoint(B::Bidiagonal{<:Real}) = Bidiagonal(B.dv, B.ev, B.uplo == 'U' ? :L : :U)
 transpose(B::Bidiagonal{<:Number}) = Bidiagonal(B.dv, B.ev, B.uplo == 'U' ? :L : :U)
+permutedims(B::Bidiagonal) = Bidiagonal(B.dv, B.ev, B.uplo == 'U' ? 'L' : 'U')
+function permutedims(B::Bidiagonal, perm)
+    Base.checkdims_perm(B, B, perm)
+    NTuple{2}(perm) == (1, 2) && return B
+    NTuple{2}(perm) == (2, 1) && return permutedims(B)
```

Comment:
There are only two options after `checkdims_perm`, right? So perhaps this can be further simplified:
```suggestion
function permutedims(B::Bidiagonal, perm)
    Base.checkdims_perm(B, B, perm)
    NTuple{2}(perm) == (2, 1) ? permutedims(B) : B
```
I have not tried it, but some chance this helps the compiler know that `false` isn't a possible return.

Reply:
Seems like there should be a method of that for `B===B` too, btw:
```
julia> x = rand(3,3); y = similar(x);

julia> @btime Base.checkdims_perm($x, $x, (2,1));
  min 4.000 ns, mean 4.116 ns (0 allocations)

julia> @eval Base function checkdims_perm(B::AbstractArray{TB,N}, perm) where {TB,N}
           indsB = axes(B)
           length(perm) == N || throw(ArgumentError("expected permutation of size $N, but length(perm)=$(length(perm))"))
           isperm(perm) || throw(ArgumentError("input is not a permutation"))
           nothing
       end
checkdims_perm (generic function with 2 methods)

julia> @btime Base.checkdims_perm($x, (2,1));
  min 1.458 ns, mean 1.570 ns (0 allocations)

julia> @btime permutedims!($y, $x, (2,1));  # so 4ns isn't nothing
  min 13.180 ns, mean 13.359 ns (0 allocations)
```